lib/request.ts

```diff
@@ -190,16 +200,18 @@ export const request = (
         }
       });
     }
-    requestObject.on("error", reject);
+    requestObject.on("error", error =>
+      reject(new RequestError(error.message, options, timings))
```

Comment:
Discussed in person: let's replace with `NetworkError`

[Documentation](https://nodejs.org/api/http.html#http_http_request_url_options_callback):

```
If any error is encountered during the request (be that with DNS resolution, TCP level errors, or actual HTTP parse errors) an 'error' event is emitted on the returned request object. As with all 'error' events, if no listeners are registered the error will be thrown.
```

Reply:
`SocketError` is a better choice IMO.

Reply:
The way I understand the documentation is that it will also include things like HTTP parse errors which for me are not really socket errors (different layer).